src/query/joining.rs

```diff
@@ -50,14 +51,17 @@ impl NestedQuery {
 /// Has Child query
 #[derive(Debug, Default, Serialize)]
 pub struct HasChildQuery {
+    #[serde(rename="type")]
     doc_type: String,
     query: Query,
     #[serde(skip_serializing_if="ShouldSkip::should_skip")]
     score_mode: Option<ScoreMode>,
     #[serde(skip_serializing_if="ShouldSkip::should_skip")]
     min_children: Option<u64>,
     #[serde(skip_serializing_if="ShouldSkip::should_skip")]
-    max_children: Option<u64>
+    max_children: Option<u64>,
+    #[serde(skip_serializing_if="ShouldSkip::should_skip")]
+    inner_hits: Option<Value>
```

Comment:
Could we also add the same to the `HasParentQuery` as the `inner_hits` field should in-theory work there too?

Reply:
Does it actually? https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-has-parent-query.html I can't find it here but in case sure, I can add `inner_hits` in there too. I will wait for a your response before to commit it just to double check I understood correctly what you mean :)

Reply:
I don't think I've ever needed it on a `has_parent` query, and it's not mentioned in the documentation for it either.  But it is mentioned here: ["Inner hits can be used by defining a inner_hits definition on a nested, has_child or has_parent query and filter"](https://www.elastic.co/guide/en/elasticsearch/reference/1.5/search-request-inner-hits.html) it might be something that's missing from the documentation elsewhere.